services/document-manager/backend/app/services/object_store_storage_service.py

```diff
@@ -72,16 +72,19 @@ def download_file(self, path, display_name, as_attachment):
         def generate(result):
             for chunk in iter(lambda: result['Body'].read(1048576), b''):
                 yield chunk
-
-        s3_response = self._client.get_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=path)
-        resp = Response(
+        try:
+            s3_response = self._client.get_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=path)
+            resp = Response(
             generate(s3_response),
             mimetype='application/pdf' if '.pdf' in display_name.lower() else 'application/zip',
             headers={
                 'Content-Disposition':
                     ('attachment; ' if as_attachment else '') + ('filename=' + display_name)
             })
-        return resp
+            
+            return resp
+        except ClientError as e:
```

Comment:
Is ClientError the only possible error type? I'm wondering if this should be a more generic exception.
Looks good though!

Reply:
I was thinking that too, there are definitely more error types but I noticed most of the other functions in this file use ClientError and in the aws boto3 doc section they mention ClientError is the general exception.

![Screenshot (167)](https://github.com/bcgov/mds/assets/127789479/af948a4b-69d3-487a-a590-c6d9841f39a8)

Reply:
The most generic exception will always just be "Exception" but if errors are handled on the FE I think it probably makes sense to let it produce a 500 if another type of error occurs (ex: if display_name gets passed in as undefined or not a string so can't do .lower() on it- errors resulting from the code or bad parameters).

Their language here is a little bit vague. They say it's "**a** general" exception, and "the most common" but they don't say it's "the gener**ic**" exception or that all errors from botocore will be ClientErrors. When I look at [this example here](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#catching-botocore-exceptions) where they catch the ClientError and then catch a different botocore exception underneath it, that indicates to me that the botocore.exceptions.ParamValidationError does not get caught as a ClientError.

Reply:
Ah I see, then would you recommend doing setting it up like this:

`except Exception as error`